src/lib/utils/hooks/useSet.ts

```diff
@@ -0,0 +1,44 @@
+import { useState } from 'react'
+
+export interface UseSetReturn<T> {
+  set: Omit<Set<T>, 'add' | 'delete' | 'clear'>
+  add: (...items: T[]) => void
+  remove: (item: T) => void
+  clear: () => void
+}
+
+/**
+ * Hook for storing an ES6 set in React state. Methods that mutate the set have been hidden via TypeScript,
+ * so if you adhere to the type definitions, the set is essentially immutable.
+ * Modifications to the set should only be done via the functions returned by the hook, which will create a new object to trigger rerendering.
+ * @param initialState
+ * @returns
+ */
+export function useSet<T>(initialState?: T[] | Set<T>): UseSetReturn<T> {
```

Comment:
Going to use this to track which rows in the new table are selected (the value is the synId of the row).